iot/iot-device-registry-infinispan/src/main/java/io/enmasse/iot/registry/infinispan/credentials/CredentialsKey.java

```diff
@@ -3,20 +3,20 @@
  * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
  */
 
-package io.enmasse.iot.registry.infinispan;
+package io.enmasse.iot.registry.infinispan.credentials;
 
+import java.io.Serializable;
 import java.util.Objects;
-import org.infinispan.protostream.annotations.ProtoDoc;
-import org.infinispan.protostream.annotations.ProtoField;
 
 /**
  * A custom class to be used as key in the backend key-value storage.
  * This uses the uniques values of a credential to create a unique key to store the credentials details.
  *
  *  See {@link CacheCredentialService CacheCredentialService} class.
  */
-@ProtoDoc("@Indexed")
-public class CredentialsKey {
+public class CredentialsKey implements Serializable {
```

Comment:
A serializable object that represents key without an equals/hashcode implementations make me suspicious.   Do we care about CredentialsKey equality anywhere or store them in Java collections?  Do these objects get serialised into storage and then recovered later?  My worry is that the built in Object.hashCode() will change  in this case.

Reply:
The class does have equals/hashCode implemented.

Reply:
sorry, my mistake.